controllers/helpers.go

```diff
@@ -434,7 +422,13 @@ func deleteCategories(client *nutanixClientV3.Client, categoryIdentifiers []*inf
 				klog.Infof("Category with value %s in category %s not found. Already deleted?", value, key)
 				continue
 			}
-			client.V3.DeleteCategoryValue(key, value)
+
+			err = client.V3.DeleteCategoryValue(key, value)
+			if err != nil {
+				errorMsg := fmt.Errorf("failed to delete category with key %s. error: %v", key, err)
+				klog.Error(errorMsg)
```

Comment:
if you're returning the error anyway, what's the point of logging? I'd recommend leaving the choice of logging the error or not to the caller.

Reply:
I did it because throughout the code, this convention has been followed. Should I change it or keep it as it is?

Reply:
Thanks for sharing.